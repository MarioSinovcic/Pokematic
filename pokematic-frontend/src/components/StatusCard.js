import React from 'react';
import ProgressBar from './ProgressBar';
import { Typography } from '@material-ui/core';
import styles from './styledComponents.css';
import Label from './Label';

function StatusCard() {
    // These should be passed in as props
    const teamName = "Team Alpha";
    const temporaryTeamLogo = "/eevee.png";
    const statusTitle = "TODO";

    return (
        <div class="StatusCardShape">
            <div className="StatusBaseShape">
            <Typography className="StatusCardTitle">{statusTitle}</Typography>
            {/* Dynamically generate cards here */}
            </div>
            <div className="StatusEdge">
                <div className="StatusCut"></div>
                <div className="StatusBottom"></div> 
            </div>
        </div>
)
}

export default StatusCard;