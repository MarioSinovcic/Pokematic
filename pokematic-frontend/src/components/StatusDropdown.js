import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import InputLabel from '@material-ui/core/InputLabel';
import FormHelperText from '@material-ui/core/FormHelperText';
import FormControl from '@material-ui/core/FormControl';
import Select from '@material-ui/core/Select';
import NativeSelect from '@material-ui/core/NativeSelect';
import { sizing } from '@material-ui/system';




function StatusDropdown() {

    return(
      <FormControl height="80%" variant="filled">

        <Select
          native
          height="20%" 
          // value={state.age}
          // onChange={handleChange}
          inputProps={{
            name: 'age',
            id: 'filled-age-native-simple',
          }}
        >
          <option className="StatusPlaceholder" value={10}>To Do</option>
          <option className="StatusPlaceholder" value={20}>In Progress</option>
          <option className="StatusPlaceholder" value={30}>In Review</option>
          <option className="StatusPlaceholder" value={30}>Done</option>

        </Select>
      </FormControl>
    )
}

export default StatusDropdown;