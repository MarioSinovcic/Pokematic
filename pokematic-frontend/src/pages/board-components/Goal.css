.team-tabs {
    display: flex;
    flex-direction: column !important;
    align-items: center;
    padding-bottom: 10px !important;
}

.sidebar-goal-name {
    align-self: flex-start !important;
    color: white;
    font-family: 'pkmn_rbygscregular' !important;
    padding-bottom: 10px;
    font-size: 10px;
}

.goal-divider {
    margin-top: 10px !important;
    width: 75% !important;
    background-color: white !important;
}

.goal-heading{
    display: flex;
    width: 100%;
    vertical-align: middle;
}

.name-width{
    width:200px;
    text-align: left;
}

.delete-goal-button{
    margin-top: 10px;
    height: 5px;
    width:20px;
    background-color: #F64747;
    border: none;
}

.delete-goal-button:hover{
    background-color: rgb(189, 59, 59);
    border: none;
}

.delete-goal-button:focus {outline:0;}

.delete-modal{
    border-radius: 10px;
    background-color: white;
    padding: 20px 20px;
}

.delete-modal-title{
    font-family: 'Roboto Mono', monospace !important;
    font-weight: 600 !important;
    font-size: 30px !important;
    line-height: 20px;
    color: #3D3D3D !important;
}

.delete-modal-message{
    font-family: 'Roboto Condensed Light', monospace !important;
    font-weight: 600 !important;
    color: rgb(145, 145, 145) !important;
    font-size: 20px !important;
    line-height: 40px;
}

.delete-modal-actions{
    display: flex !important;
    justify-content: center;

}

.cancel-delete{
    font-family: 'Roboto Condensed', monospace !important;
    right: 200px;
}

.confirm-delete{
    left: 200px;
    background-image: url('../../images/delete.png');
    background-repeat: no-repeat;
    background-color: #ffffff;
    background-size: 30px 30px;
    background-position: center;
    height: 40px;

    padding: 10px 10px;
    border-radius: 10px;
    outline: none;
    border: none;
}