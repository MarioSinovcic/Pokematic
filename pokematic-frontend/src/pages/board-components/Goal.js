import React from 'react';
import '../Board.css';
import { Typography, Divider, ListItem } from '@material-ui/core';
import ProgressBar from '../../shared-components/ProgressBar';

function Goal(props) {

  function calculateProgress(){
    return (props.progress / props.experiencePoints *100)
  }

    return (
        <div className="TeamTabs">
            <ListItem button key={props.text} className="TeamTabs">
              <Typography className="TaskName">{props.name}</Typography>
              <ProgressBar progress={calculateProgress()} />
            </ListItem>
            <Divider className="GoalDivider" /> 
        </div>
)
}

export default Goal;