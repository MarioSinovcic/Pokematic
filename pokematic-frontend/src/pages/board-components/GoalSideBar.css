.SideBar {
    min-width: 600px;
    flex-shrink: 0;
    background-color: black !important;
    box-shadow: #3D3D3D;
    position: absolute;
    z-index: -1;
}

.MuiPaper-root {
    background-color: #3D3D3D !important;
}

.MuiList-padding {
    padding-bottom: 8px;
}

.TaskFilter {
    font-family: 'Roboto Condensed Bold', sans-serif !important;
    font-weight: 600 !important;
    align-self: center !important;
    text-align: center !important;
    color: white;
}

.AllTasks {
    padding-top: 70px;
    width: 225px;
    font-size: 35px !important; 
}

.MyTasks {
    font-size: 25px !important; 
}

.SideBarDivider {
    background: rgba(255, 255, 255, 0.2) !important;
    padding: 3px;
}

.GoalDivider {
    margin-top: 10px !important;
    width: 75% !important;
    background-color: white !important;
}

.TeamTabs {
    display: flex;
    flex-direction: column !important;
    align-items: center;
    padding-bottom: 10px !important;
}

.TaskName {
    align-self: flex-start !important;
    color: white;
    font-family: 'pkmn_rbygscregular' !important;
    padding-bottom: 10px;
    font-size: 10px;
}

.NewGoalButton{
    padding-left: 20px;
    top: 76%;
    z-index: 20;
    position: absolute;
}