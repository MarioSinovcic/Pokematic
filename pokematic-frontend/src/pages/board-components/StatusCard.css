.StatusCardShape {
    display: flex;
    flex-direction: row;
    width: 300px;
    height: 750px;
    padding: 25px;
    
}

.StatusHeaders {
    display: flex;
    flex-direction: row;
    justify-content: space-between
}

.StatusEdge {
    display: flex;
    flex-direction: column;
}

.StatusBaseShape {
    width: 300px;
    height: 750px;
    background: #E7E7E7;
    /* box-shadow: 5px 5px 5px grey; */
    /* position: absolute; */
}

.StatusBottom {
    width: 64px;
    height: 750px;
    background:#E7E7E7;
    /* position: relative; */
    /* top:10px;
    right:300px; */
    /* box-shadow: 5px 5px 5px grey; */

}

.StatusCut {
    height: 0px;
    /* position: relative; */
    /* bottom: 69px;
    right: 300px; */
    border-right: 64px solid transparent;
    border-bottom: 61px solid #E7E7E7;
}

.StatusCardTitle {
    margin: 56px 0 0 30px !important;
    font-family: 'Roboto Condensed', sans-serif !important;
    font-size: 20px !important;
    font-weight: 600 !important;
    color: #3D3D3D !important;
}

.Tasks {
    position: absolute;
    padding: 10px 18px 10px 25px;
    overflow-y: auto;
    height: 610px !important;

}

#scrollbar::-webkit-scrollbar-track
{
	border-radius: 5px;
	background-color: transparent;
}

#scrollbar::-webkit-scrollbar
{
	width: 12px;
	background-color: transparent;
}

#scrollbar::-webkit-scrollbar-thumb
{
	/* border-radius: 5px; */
	background-color: #C4C4C4;
}
