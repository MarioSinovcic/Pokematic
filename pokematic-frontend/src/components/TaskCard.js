import React from 'react';
import ProgressBar from './ProgressBar';
import { Typography } from '@material-ui/core';
import styles from './styledComponents.css';
import Label from './Label';
import StatusDropdown from './StatusDropdown';
import { sizing } from '@material-ui/system';

function TaskCard() {
    // These should be passed in as props
    const taskID = "34"
    const taskTitle = "Implement collaboration feature";


    return (
        <div class="TaskCard">
            <div className="TaskHeaders">
            <Typography className="TaskTitleText">{taskTitle}</Typography>
                <Typography className="TaskIDText">#{taskID}</Typography>
            </div>
            <div className="TaskLabels">
                <Label />
            <StatusDropdown />

            </div>
            
        </div>
)
}

export default TaskCard;