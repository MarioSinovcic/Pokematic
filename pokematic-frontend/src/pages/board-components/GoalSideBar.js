import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Drawer from '@material-ui/core/Drawer';
import List from '@material-ui/core/List';
import Typography from '@material-ui/core/Typography';
import Divider from '@material-ui/core/Divider';
import ListItem from '@material-ui/core/ListItem';
import AddIcon from '@material-ui/icons/Add';
import Goal from './Goal';
import ModalButton from '../../shared-components/ModalButton';
import './GoalSideBar.css'

const drawerWidth = 250;

const useStyles = makeStyles(theme => ({
  root: {
    display: 'flex',
  },
  appBar: {
    zIndex: theme.zIndex.drawer + 1,
  },
  drawer: {
    width: drawerWidth,
    flexShrink: 0,
  },
  drawerPaper: {
    width: drawerWidth,
  },
  content: {
    flexGrow: 1,
    padding: theme.spacing(3),
  },
  // necessary for content to be below app bar
  toolbar: theme.mixins.toolbar,
}));


export class GoalSideBar extends React.Component {

  render (){
    const classes = useStyles;
    const newModalIcon = <AddIcon style={{fontSize: "35px", color: "#3D3D3D"}}/>;

    if (!this.props.goalsList){ 
      return <div>laoding</div>;
    }
    else{         
        var goalsToRender = this.props.goalsList.map((goalData) => 
            <Goal 
                key={goalData["id"]} //not used: just avoiding warnings
                id={goalData["id"]} 
                name={goalData["name"]} 
                experiencePoints={goalData["experiencePoints"]} 
                progress={goalData["progress"]} 
                />
          )
      return(
        <div className={classes.root}>
          <Drawer
            className="SideBar"
            variant="permanent"
            classes={{
              paper: classes.drawerPaper,
            }}
          >
            <div className={classes.toolbar} />
            <List>
                <ListItem button key={"All Tasks"} className="TeamTabs TaskButton">
                <Typography className="TaskFilter AllTasks">ALL TASKS</Typography>
                </ListItem>

                <ListItem button key={"My Tasks"} className="TeamTabs">
                <Typography className="TaskFilter MyTasks">MY TASKS</Typography>
                </ListItem>
            </List>
            <Divider className="SideBarDivider"/>
            <List>
              {goalsToRender}
            </List>
            <div className="NewGoalButton">
              <ModalButton  icon={newModalIcon} theme="light" type="new-goal"/>
            </div>
          </Drawer>
        </div>
      );
    }
  }
}

export default GoalSideBar;