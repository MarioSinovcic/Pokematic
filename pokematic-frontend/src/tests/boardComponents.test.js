import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { createStore } from 'redux';
import rootReducers from '../redux/reducers';
import Goal from '../pages/board-components/Goal';
import GoalSideBar from '../pages/board-components/GoalSideBar';
import StatusCard from '../pages/board-components/StatusCard';
import StatusDropdown from '../pages/board-components/StatusDropdown';
import TaskCard from '../pages/board-components/TaskCard';



describe('board components', () => {

    let store;
    beforeEach(() => {
        store = createStore(rootReducers)
    })

    describe('snapshot matching', () => {

        it('Goal matches snapshot' , () => {
            const goal = renderer.create(<Provider store={store}>
                <Goal />
            </Provider>).toJSON();
            expect(goal).toMatchSnapshot();
        });

        it('Goal Side Bar matches snapshot' , () => {
            const goalSidebar = renderer.create(<Provider store={store}>
                <GoalSideBar />
            </Provider>).toJSON();
            expect(goalSidebar).toMatchSnapshot();
        });

        it('Status Card matches snapshot' , () => {
            const statusCard = renderer.create(<Provider store={store}>
                <StatusCard />
            </Provider>).toJSON();
            expect(statusCard).toMatchSnapshot();
        });

        it('Status DropDown matches snapshot' , () => {
            const status = `In Progress`;

            const dropdown = renderer.create(<Provider store={store}>
                <StatusDropdown currentStatus={status}/>
            </Provider>).toJSON();
            expect(dropdown).toMatchSnapshot();
        });

        it('Taskcard matches snapshot' , () => {
            const task = renderer.create(<Provider store={store}>
                <TaskCard />
            </Provider>).toJSON();
            expect(task).toMatchSnapshot();
        });

    })

});