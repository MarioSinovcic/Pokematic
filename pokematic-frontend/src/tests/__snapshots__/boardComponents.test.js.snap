// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`board components snapshot matching Goal Side Bar matches snapshot 1`] = `
<div>
  laoding
</div>
`;

exports[`board components snapshot matching Goal matches snapshot 1`] = `
<div>
  <div
    className="team-tabs"
  >
    <li
      className="MuiListItem-root team-tabs MuiListItem-gutters"
      disabled={false}
    >
      <div
        className="goal-heading"
      >
        <p
          className="MuiTypography-root name-width sidebar-goal-name MuiTypography-body1"
        />
        <button
          className="delete-goal-button"
          onClick={[Function]}
        />
      </div>
      <div
        className="progress-bar"
      >
        <div
          aria-valuenow={NaN}
          className="MuiLinearProgress-root MuiLinearProgress-colorPrimary goal-progress-bar MuiLinearProgress-determinate"
          role="progressbar"
        >
          <div
            className="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Determinate"
            style={
              Object {
                "transform": "translateX(NaN%)",
              }
            }
          />
        </div>
      </div>
    </li>
    <hr
      className="MuiDivider-root goal-divider"
    />
  </div>
  <div />
</div>
`;

exports[`board components snapshot matching Status Card matches snapshot 1`] = `null`;

exports[`board components snapshot matching Status DropDown matches snapshot 1`] = `
<div
  className="MuiFormControl-root makeStyles-dropDown-86"
>
  <div
    className="MuiInputBase-root MuiInput-root makeStyles-dropDownMenu-87 MuiInputBase-formControl MuiInput-formControl"
    onClick={[Function]}
  >
    <div
      aria-haspopup="listbox"
      aria-labelledby="demo-simple-select-outlined-label demo-simple-select-outlined"
      className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
      id="demo-simple-select-outlined"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      role="button"
      tabIndex={0}
    >
      In Progress
    </div>
    <input
      onAnimationStart={[Function]}
      type="hidden"
      value="In Progress"
    />
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root MuiSelect-icon makeStyles-icon-89"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
    </svg>
  </div>
</div>
`;

exports[`board components snapshot matching Taskcard matches snapshot 1`] = `
<div>
  <div
    className="TaskCard"
  >
    <div
      className="TaskHeaders"
      onClick={[Function]}
    >
      <p
        className="MuiTypography-root TaskTitleText MuiTypography-body1"
      />
      
    </div>
    <div
      className="TaskLabels"
    >
      <div
        id="Label"
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-text team-label pokemon-label"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          style={
            Object {
              "backgroundColor": "#F64747",
            }
          }
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            undefined XP
          </span>
        </button>
      </div>
      <div
        className="MuiFormControl-root makeStyles-dropDown-86"
      >
        <div
          className="MuiInputBase-root MuiInput-root makeStyles-dropDownMenu-87 MuiInputBase-formControl MuiInput-formControl"
          onClick={[Function]}
        >
          <div
            aria-haspopup="listbox"
            aria-labelledby="demo-simple-select-outlined-label demo-simple-select-outlined"
            className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
            id="demo-simple-select-outlined"
            onBlur={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            role="button"
            tabIndex={0}
          >
            <span
              dangerouslySetInnerHTML={
                Object {
                  "__html": "&#8203;",
                }
              }
            />
          </div>
          <input
            onAnimationStart={[Function]}
            type="hidden"
          />
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root MuiSelect-icon makeStyles-icon-89"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
  <div />
</div>
`;
